src/tests/krate.rs

```diff
@@ -1447,6 +1447,251 @@ fn yank_max_version() {
+            .version("0.2.0")
+            .expect_build(&conn);
+    }
+
+    // Verify that crates start without any documentation so the next assertion can *prove*
+    // that it was the one that added the documentation
+    {
+        let mut req = ::req(app.clone(), Method::Get, "/api/v1/crates/docscrate");
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: CrateResponse = ::json(&mut response);
+        assert_eq!(json.krate.documentation, None);
+    }
+
+    // Make sure publishing a version less than the max version doesn't add documentation
+    {
+        let mut req = ::new_req_with_documentation(
+            app.clone(),
+            "docscrate",
+            "0.1.0",
+            Some("http://goo.gl".to_owned()),
+        );
+        ::sign_in_as(&mut req, &user);
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: GoodCrate = ::json(&mut response);
+        assert_eq!(json.krate.documentation, Some("http://goo.gl".to_owned()));
+    }
+
+    {
+        let mut req = ::req(app.clone(), Method::Get, "/api/v1/crates/docscrate");
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: CrateResponse = ::json(&mut response);
+        assert_eq!(json.krate.documentation, None);
+    }
+
+    // 2. Add documentation
+    {
+        let mut req = ::new_req_with_documentation(
+            app.clone(),
+            "docscrate",
+            "0.2.1",
+            Some("http://foo.rs".to_owned()),
+        );
+        ::sign_in_as(&mut req, &user);
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: GoodCrate = ::json(&mut response);
+        assert_eq!(json.krate.documentation, Some("http://foo.rs".to_owned()));
+    }
+
+    // Ensure latest version also has the same documentation
+    {
+        let mut req = ::req(app.clone(), Method::Get, "/api/v1/crates/docscrate");
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: CrateResponse = ::json(&mut response);
+        assert_eq!(json.krate.documentation, Some("http://foo.rs".to_owned()));
+    }
+
+    // Make sure removing the documentation in a version less than the max version doesn't overwrite anything
+    {
+        let mut req = ::new_req_with_documentation(app.clone(), "docscrate", "0.1.1", None);
+        ::sign_in_as(&mut req, &user);
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: GoodCrate = ::json(&mut response);
+        assert_eq!(json.krate.documentation, None);
+    }
+
+    // Ensure latest version still has the same documentation
+    {
+        let mut req = ::req(app.clone(), Method::Get, "/api/v1/crates/docscrate");
+        let mut response = ok_resp!(middle.call(&mut req));
+        let json: CrateResponse = ::json(&mut response);
+        assert_eq!(json.krate.documentation, Some("http://foo.rs".to_owned()));
+    }
+
+    // 3. Remove the documentation
```

Comment:
And again, imo lines 1637-1691 are not providing any additional coverage. Please remove, thank you!